File: src/main/java/co/uk/jiveelection/campaign/utils/RegexEntity.java
package co.uk.jiveelection.campaign.utils;

import com.google.auto.value.AutoValue;

/**
 * Describes information about a regex match to be accessed outside a matcher loop.
 */
@AutoValue
public abstract class RegexEntity {
    public static RegexEntity create(int start, int end, String matchedText, String replacementText) {
        return new AutoValue_RegexEntity(start, end, matchedText, replacementText);
    }

    public abstract int start();

    public abstract int end();

    public abstract String matchedText();

    public abstract String replacementText();
}

File: src/main/kotlin/co/uk/jiveelection/campaign/utils/RegexEntity.kt
package co.uk.jiveelection.campaign.utils

/**
 * Describes information about a regex match to be accessed outside a matcher loop.
 */

data class RegexEntity(val start: Int, val end: Int, val matchedText: String, val replacementText: String) {

    companion object {
        fun create(start: Int, end: Int, matchedText: String, replacementText: String): RegexEntity {
            return RegexEntity(start, end, matchedText, replacementText)
        }
    }
}
